### PR TITLE
Test adapter `Validate::isUnsignedInt` 

### PR DESCRIPTION
**Pull request by PululuK**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop 
| Description?      | Improve and Add test in Adapter `ValidateTest`. Needed for another PR
| Type?             | improvement
| Category?         | TE 
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | N/A
| How to test?      | Tests green ;)
| Possible impacts? | N/A

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/24629)

### CODE REVIEW

File: tests/Unit/Adapter/ValidateTest.php [hunk 1]
```diff
@@ -33,12 +33,25 @@
 
 class ValidateTest extends TestCase
 {
+    private $validate;
```
Comment (Progi1984) [inline tests/Unit/Adapter/ValidateTest.php:39]:
```suggestion
    /**
     * @var Validate
     */
    private $validate;
```

**Review by Progi1984 — APPROVED** [review]
A small feedback but I commited it ;)

File: tests/Unit/Adapter/ValidateTest.php [hunk 2]
```diff
@@ -33,12 +33,28 @@
...
+        $this->validate = new Validate();
+    }
+
     /**
      * @dataProvider getIsOrderWay
      */
     public function testIsOrderWay(int $expected, $input): void
     {
-        self::assertEquals($expected, Validate::isOrderWay($input));
+        self::assertEquals($expected, $this->validate::isOrderWay($input));
```
Comment (matks) [inline tests/Unit/Adapter/ValidateTest.php:57]: Hi! I'm sure you did this on purpose but I don't understand why.

Why
```
self::assertEquals($expected, $this->validate::isOrderWay($input));
```
is better than
```
self::assertEquals($expected, Validate::isOrderWay($input));
```

Comment (PululuK) [reply tests/Unit/Adapter/ValidateTest.php:57]: @matks `Validate` has been injected in constructor. To keep consistent we need to use this one

Comment (matks) [reply tests/Unit/Adapter/ValidateTest.php:57]: Oh! I understand now. The root issue is that adapter Validate class still using static calls, that directly relate to unnamespaced class.

What a very weird class `PrestaShop\PrestaShop\Adapter\Validate` 😅

**Review by matks — APPROVED** [review]

**Comment by Progi1984** [comment]
Thanks @PululuK

**Pull request merged**